Command to run build

csrwng opened the pull request:
Adds a command that takes a build json/yaml and runs the build to completion. It assumes that multiple clients that start the same build also use the same name for the build. If a build exists with that name, then that build is followed.

csrwng commented:
@stevekuznetsov

stevekuznetsov commented on tools/release-ci/pkg/buildrunner/runbuild.go:
```diff
@@ -0,0 +1,211 @@
+
+var (
+	WaitForRunningTimeout = 10 * time.Minute
+	MaximumBuildDuration  = 4 * time.Hour
```
really?

csrwng replied:
Yes, seems like a good upper bound

stevekuznetsov commented on tools/release-ci/pkg/buildrunner/runbuild.go:
```diff
@@ -0,0 +1,211 @@
+}
+
+func extractSuccessfulBuildName(str string) (string, error) {
+	createdBuildRegex := regexp.MustCompile("build \"([^\"]*)\" created")
```
can't we `-o name` to not need to parse potentially unstable human readable text?

csrwng replied:
We can parse the Yaml/JSON and get the name from there. I guess I was being lazy? I don't want to make an extra call to the server if I don't need to.

csrwng replied:
Will fix

stevekuznetsov commented on tools/release-ci/pkg/buildrunner/runbuild.go:
```diff
@@ -0,0 +1,211 @@
+	if err != nil {
+		return err
+	}
+	for {
```
you can `time.Tick` in the `for`

csrwng replied:
will update

stevekuznetsov commented on tools/release-ci/pkg/buildrunner/runbuild.go:
```diff
@@ -0,0 +1,211 @@
+
+func checkBuildSuccess(name string) error {
+	status, err := getBuildStatus(name)
+	if !isComplete(status) {
```
isn't this a logic error?

stevekuznetsov replied:
based on when we call this func

csrwng replied:
No, following the build doesn't guarantee that the build will be done running. I've made that wrong assumption before in pipelines. There's all kinds of reasons why the `logs -f` command can terminate early and I really want to make sure that the build is done.

stevekuznetsov replied:
ack

stevekuznetsov commented on tools/release-ci/pkg/buildrunner/runbuild.go:
```diff
@@ -0,0 +1,211 @@
+	if err != nil {
+		return err
+	}
+	for {
```
also this -- shouldn't it get eaten by `oc logs -f`?

csrwng replied:
see above

stevekuznetsov commented:
Looks good

csrwng commented:
@stevekuznetsov addressed your comments except for the one about using Tick. It made the code more complex and I didn't see much gained by it. I added the YAML parser to the vendor directory and glide decided to update the other deps. If you think we're better off not updating those, I can look into only adding the yaml parser.

stevekuznetsov commented:
LGTM
